### PR TITLE
CORE-8862: Prevent creation of a second group in a mutual TLS cluster

### PR DESCRIPTION
**Pull request by yift-r3**
In mutual TLS, a user should not be able to create a second group in a cluster. This PR prevent this.

(the user can still create two groups and then change the TLS type, but this will break a lot more and will be handle in the future)

### CODE REVIEW

File: libs/virtual-node/virtual-node-common/src/main/kotlin/net/corda/libs/virtualnode/common/exception/AnotherGroupExistsMutualTlsException.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package net.corda.libs.virtualnode.common.exception
+
+import net.corda.v5.base.exceptions.CordaRuntimeException
+
+/**
+ * Another group exists in the cluster in mutual TLS mode
+ */
+class AnotherGroupExistsMutualTlsException :
+    CordaRuntimeException("Another group already exists in the cluster in mutual TLS.")
```
Comment (dimosr) [inline libs/virtual-node/virtual-node-common/src/main/kotlin/net/corda/libs/virtualnode/common/exception/AnotherGroupExistsMutualTlsException.kt]:
```suggestion
    CordaRuntimeException("Another group already exists in the cluster configured in mutual TLS mode.")
```

File: libs/virtual-node/virtual-node-datamodel/src/integrationTest/kotlin/net/corda/libs/configuration/datamodel/tests/VirtualNodeRepositoryTest.kt [hunk 2]
```diff
@@ -194,4 +198,61 @@ class VirtualNodeRepositoryTest {
         assertThat(changedEntity).isNotNull
         assertThat(changedEntity!!.state).isEqualTo(VirtualNodeState.IN_MAINTENANCE)
     }
+
+    @Nested
+    inner class OtherGroupsExistsTest {
+        @AfterEach
+        @BeforeEach
+        fun cleanUp() {
+            entityManagerFactory.createEntityManager().use { em ->
```
Comment (dimosr) [inline libs/virtual-node/virtual-node-datamodel/src/integrationTest/kotlin/net/corda/libs/configuration/datamodel/tests/VirtualNodeRepositoryTest.kt:207]: In the interest of keeping that test a bit minimal and reducing the duration (given they are integration tests), would it make sense to move the creation of vnodes on the `BeforeEach` and then have 2 tests, one testing that otherGroupsExists returns `false` when providing the created group and `true` when providing another (non-existing) group? Seems like the third test case currently is similar to second one in my proposed ones.

**Comment by yift-r3** [comment]
Do not review.

We have decided to document the behaviour of multiple groups instead of preventing the creation of one.

**Review by williamvigorr3 — APPROVED** [review]
LGTM (When @dimosr is happy).

**Comment by dimosr** [comment]
Closing as per Yiftach's comment above

**Pull request closed without merging**